components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/db/TabEntity.kt

```diff
@@ -49,13 +48,13 @@ internal data class TabEntity(
     @ColumnInfo(name = "created_at")
     var createdAt: Long
 ) {
-    internal fun getStateFile(context: Context): AtomicFile {
-        return AtomicFile(File(getStateDirectory(context), stateFile))
+    internal fun getStateFile(filesDir: File): AtomicFile {
+        return AtomicFile(File(getStateDirectory(filesDir), stateFile))
```

Comment:
I'm not sure about this refactoring. The API was meant to give you a specific file and the context was needed to lookup a directory. Now this is part of the public API and every caller needs to know which directory to pass in - and its possible to pass in other/wrong directories and files.